src/shared/sleep-config.c

```diff
@@ -127,6 +133,50 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+static int acquire_bus(sd_bus **bus) {
+        int r;
+
+        assert(bus);
+
+        if (*bus)
+                return 0;
+
+        r = bus_connect_transport(BUS_TRANSPORT_LOCAL, NULL, false, bus);
+        if (r < 0)
+                return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL);
+
+        return 0;
+}
+
+/* Freeze or thaw user.slice */
+int freeze_thaw_user_slice(const char **method) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *m = NULL;
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        sd_bus *bus;
+        int r;
+
+        if (*method == NULL)
+                return -1;
+
+        r = acquire_bus(&bus);
+        if (r < 0)
+                return r;
+
+        r = bus_message_new_method_call(bus, &m, bus_systemd_mgr, *method);
```

Comment:
just use `sd_bus_call_method` which wraps all 3 calls in one

Reply:
ok